### ISSUE TITLE
Duplicate papers collected by auto-updater

### ISSUE THREAD

**Issue by dongbohu** [greenelab/preprint-similarity-search#98]
@danich1: when cleaning up the wonky papers whose number of tokens is less than 20, I found another issue that is a little annoying: 518 duplicate papers are collected by the auto-updater by accident. The reason is because some journals' names in the pre-2020 `pmc_oa_file_list.tsv` were slightly different from the journal names stored in tarball files that were downloaded from NIH FTP site. 

One example is `PMC2715451`. In the pre-2020 `pmc_oa_file_list.tsv`, it is recorded as:
```
comm_use.I-N.xml.tar.gz	Microbiology/PMC2715451.nxml
```
But in `comm_use.I-N.xml.tar.gz` that was downloaded from NIH FTP site, this paper was stored in	the path of:
```
Microbiology_(Reading)/PMC2715451.nxml
```
Because the auto-updater read `pmc_oa_file_list.tsv` first and used `Microbiology/PMC2715451.nxml` as the key to distinguish whether a paper has been processed before, `Microbiology_(Reading)/PMC2715451.nxml` was considered as a new paper.

Two things that I think should be done to fix it:
1. The duplicate papers should be removed from both new embeddings and new token_counter files. 
2. The auto-updater should use the PMC id only w/o the journal name (in the above example, `PMC2715451`) to decide whether a paper has been processed.

What do you think?

**Comment by dongbohu** [comment]
By the way, I also compared the feature vectors of these duplicate papers in `paper_dataset_full.tsv`. They are not the same, but relatively close. Of the 518 pairs of duplicates, the maximum distance is 1.133, and the minimum distance is 0.0825.

**Comment by danich1** [comment]
>Two things that I think should be done to fix it:
    1. The duplicate papers should be removed from both new embeddings and new token_counter files.
    2. The auto-updater should use the PMC id only w/o the journal name (in the above example, PMC2715451) to decide whether a paper has been processed.
What do you think?

Wow that's annoying. Yeah I agree that PMC id matching makes the most sense.

>By the way, I also compared the feature vectors of these duplicate papers in paper_dataset_full.tsv. They are not the same, but relatively close. Of the 518 pairs of duplicates, the maximum distance is 1.133, and the minimum distance is 0.0825.

Interesting it's probably slight version changes within these documents, but nevertheless the duplicates should be removed.

**Comment by dongbohu** [comment]
@danich1 I have removed 3,917 wonky papers (papers whose number of token counts is less than 20) from the following files in the pre-2020 collection:
- global_doc_word_counter.tsv
- paper_dataset_full.tsv
- paper_dataset_tsne_square.tsv

Since 3,917 rows have been removed from `paper_dataset_tsne_square.tsv`, do we need to recalculate the `dim1`, `dim2` and `squarebin_id` columns in this file? If yes, how to calculate them? If not, I will merge these files with the new papers collected by auto-updater and generate the final collection. Thanks.

**Comment by dongbohu** [comment]
Another question for @danich1: Do we need to recalculate `pca_components.tsv` because some papers have been removed from pre-2020 collection?

**Comment by dongbohu** [comment]
> 1. Call the generate_SAUCIE_coordinates from the auto updater pipeline using the pre-2020 files as arguments

When `generate_SAUCIE_coordinates()` is called like this:
```python
generate_SAUCIE_coordinates(new_papers, paper_landscape_file, paper_landscape_json_file)
```
I suppose `new_papers` will be the pre-2020 `paper_dataset_full.tsv` that has wonky papers removed, and `paper_landscape_file` will be a blank `paper_dataset_tsne_square.tsv` without any data row, right?

**Comment by danich1** [comment]
Good call. Yes paper landscape file should be blank when calling the function. One temp fix you will need to do is change the following lines or add a switch statement:

```python
# Grab the SAUCIE updated coordinates
    landscape_df = pd.read_csv(f"{paper_landscape_file}", sep="\t").append(
        pd.DataFrame(coordinates, columns=["dim1", "dim2"]).assign(
            document=new_papers_df.document.tolist(),
            journal=new_papers_df.journal.tolist(),
        )[["dim1", "dim2", "journal", "document"]],
        sort=True,
    )
```

should be 

```python
# Grab the SAUCIE updated coordinates
    landscape_df =  pd.DataFrame(coordinates, columns=["dim1", "dim2"]).assign(
            document=new_papers_df.document.tolist(),
            journal=new_papers_df.journal.tolist(),
        )
```

Forgot that my code works with an already generated mapper. This will change to work without mapped data (not that I think the function will be called again without mapped data present)

**Comment by dongbohu** [comment]
Turned out that the square bins are quite different after the wonky papers are removed. Before the removal there were 512 bins, and after removal there are only 499 bins. Does this make sense to you, @danich1?

**Comment by danich1** [comment]
>Turned out that the square bins are quite different after the wonky papers are removed. Before the removal there were 512 bins, and after removal there are only 499 bins. Does this make sense to you, @danich1?

Sounds about right given that the outliers are removed. There were probably a few papers that needed to be removed that I didn't realize were in the dataset.
